vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideExternalLinkManager.java

```diff
@@ -0,0 +1,79 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.SlideExternalLinkRepository;
+import edu.asu.diging.vspace.core.model.IExternalLinkSlide;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.ExternalLinkSlide;
+import edu.asu.diging.vspace.core.services.ISlideExternalLinkManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+
+@Transactional
+@Service
+public class SlideExternalLinkManager
+        implements ISlideExternalLinkManager {
+
+    @Autowired
+    private ISlideManager slideManager;
+
+    @Autowired
+    private SlideExternalLinkRepository externalLinkRepo;
+
+    @Override
+    public IExternalLinkSlide getLink(String externalLinkID) {
+        Optional<ExternalLinkSlide> externalLink = externalLinkRepo.findById(externalLinkID);
+        if (externalLink.isPresent()) {
+            return externalLink.get();
+        }
+        return null;
+    }
+    
+    @Override
+    public void addToLinkList(ISlide slide, IExternalLinkSlide link) {
+        slide.getExternalLinks().add(link);
+    }
+
+    @Override
+    public void removeFromLinkList(ISlide slide, IExternalLinkSlide link) {
+        slide.getExternalLinks().remove(link);
+    }
+
+    @Override
+    public void deleteLink(String linkId) {
+        IExternalLinkSlide link = getLink(linkId);
+        removeFromLinkList(link.getSlide(), link);
+        externalLinkRepo.delete((ExternalLinkSlide) link);
+    }
+
+    @Override
+    public IExternalLinkSlide createExternalLink(String label, String url, String slideId) {
+        ISlide slide = slideManager.getSlide(slideId);
+        IExternalLinkSlide externalLink = new ExternalLinkSlide();
+        externalLink.setExternalLink(url);
+        externalLink.setLabel(label);
+        externalLink.setSlide(slide);
+        externalLinkRepo.save((ExternalLinkSlide) externalLink);
+        slide.getExternalLinks().add(externalLink);
+        slideManager.updateSlide(slide);
```

Comment:
this seems not the right place to do this. I would create the external link here and return it (without saving), then where it is called, add the link to the slide and save the slide (the saving should cascade to the link object so the link object is also saved).